src/hooks/useDataFrameCache.tsx

```diff
@@ -0,0 +1,103 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import type { DataFrameCache } from '../helpers/dataframe/cache.js'
+import { createDataFrameCache } from '../helpers/dataframe/cache.js'
+import type { DataFrame, Fetch, ResolvedValue } from '../helpers/dataframe/types.js'
+
+export interface UseDataFrameCacheOptions {
+  cacheKey?: string
+  numRows: number
+  columnDescriptors: { name: string, sortable?: boolean, metadata?: any }[]
+  fetch?: Fetch
+  getRowNumber?: (row: number) => ResolvedValue<number> | undefined
+}
+
+/**
+ * Hook to create a complete DataFrame with built-in caching and automatic re-rendering.
+ *
+ * @param options Configuration options
+ * @param options.cacheKey Optional key to identify the cache. When the key changes, a new cache is created.
+ * @param options.numRows Number of rows in the DataFrame
+ * @param options.columnDescriptors Column definitions for the DataFrame
+ * @param options.fetch Optional async function to fetch data
+ * @param options.getRowNumber Optional custom getRowNumber implementation. If not provided, defaults to row index.
+ * @returns Object containing a ready-to-use DataFrame and helper functions
+ */
+export function useDataFrameCache(options: UseDataFrameCacheOptions = { numRows: 0, columnDescriptors: [] }) {
+  const { cacheKey, numRows, columnDescriptors, fetch, getRowNumber: customGetRowNumber } = options
+
+  // Use a ref to store the cache so it persists across re-renders but can be replaced when cacheKey changes
+  const cacheRef = useRef<{ key: string | undefined, cache: DataFrameCache, version: number } | null>(null)
```

Comment:
not sure of my comment, but... instead of relying on a React ref + a hook, maybe we could transform this file into a simpler function that transform a DataFrame into a cached DataFrame (as we had initially in hightable), optionally passing a cache as an argument/option? The optional cache could still be stored in a React state, or Ref, but it would decouple a bit more.